helpers/hypervisor/apis/kvm/sdk.py

```diff
@@ -674,6 +674,159 @@ def create_vm(self, name, vcpus, ram, disks, cdrom_iso=None, os_type=None, os_va
+            'disable_root: False',
+            'password: {0}'.format(root_password),
+            'ssh_pwauth: True',
+            'chpasswd:',
+            '  list: |',
+            '    root:{0}'.format(root_password),
+            '    ubuntu:{0}'.format(root_password),
+            '  expire: False',
+            'runcmd:',
+            '  - [sed, -ie, "s/PermitRootLogin prohibit-password/PermitRootLogin yes/", /etc/ssh/sshd_config]',
+            '  - [sed, -ie, "s/PasswordAuthentication no/PasswordAuthentication yes/", /etc/ssh/sshd_config]',
+            '  - [service, ssh, restart]',
+            ''
+        ]
+
+        # Check if vm already exists with this name
+        vm = None
+
+        try:
+            vm = self._conn.lookupByName(name)
+        except libvirt.libvirtError:
+            pass
+
+        if vm and force:
+            self.delete_vm(vm, True)
+        elif vm and not force:
+            raise Exception('VM {0} is still defined on this hypervisor. Use the force=True option to delete.'.format(name))
+
+        if self.ssh_client.dir_exists(vm_directory):
+            exists, used_disk, vm_name = self._check_disks_in_use([ci_iso, boot_disk])
+            if exists:
+                raise Exception("Virtual Disk {0} in used by {1}".format(used_disk, vm_name))
+
+            self.ssh_client.dir_delete(vm_directory)
+
+        self.ssh_client.dir_create(vm_directory)
+        # Copy template image
+        self.ssh_client.run(["cp", qcow_file, boot_disk])
+
+        # Resize image
+        self.ssh_client.run(["qemu-img", "resize", boot_disk, boot_disk_size])
+
+        # Create metadata and user data file
+        self.ssh_client.file_write(meta_data, '\n'.join(meta_data_lines))
+
+        self.ssh_client.file_write(user_data, '\n'.join(user_data_lines))
+
+        # Generate iso for cloud-init
+        self.ssh_client.run(["genisoimage", "-output", ci_iso, "-volid", "cidata", "-joliet", "-r", user_data, meta_data])
+
+        # Create extra disks
+        extra_disks = []
+        if amount_disks > 0 and size > 0:
+            if not self.ssh_client.dir_exists(mountpoint):
+                raise Exception("Directory {0} doesn't exists.".format(mountpoint))
+
+            for i in xrange(1, amount_disks+1):
+                disk_path = "%s/%s_%02d.qcow2" % (mountpoint, name, i,)
+                exists, used_disk, vm_name = self._check_disks_in_use(disk_path)
+                disk_exists_filesystem = self.ssh_client.file_exists(disk_path)
+                if disk_exists_filesystem and exists:
+                    raise Exception("Virtual Disk {0} in used by {1}".format(used_disk, vm_name))
+                elif disk_exists_filesystem:
+                    self.ssh_client.file_delete(disk_path)
+
+                self.ssh_client.run(['qemu-img', 'create', '-f', 'qcow2', disk_path, size])
+                extra_disks.append(disk_path)
+
+        all_disks = [{'mountpoint': boot_disk, "format": "qcow2", "bus": "virtio"}]
```

Comment:
No need for the `extra_disks` list
Just declare this `boot_disk` line above the `amount_disk` loop and append the `extra_disk` right away